source/ui/AMBeamlineCameraWidget.cpp

```diff
@@ -141,7 +142,12 @@ void AMBeamlineCameraWidget::onVideoWidgetDoubleClicked(const QPointF &clickPoin
 void AMBeamlineCameraWidget::onRecenterCrosshairPushbuttonClicked()
 {
 	if(!crosshairLocked_)
-		setCrosshairPosition(crosshairCenter_);
+	{
+
+		double x = QInputDialog::getDouble(this, "Crosshair Position", "X coordinate(0-1):", crosshairCenter_.x(),0,1,4);
```

Comment:
They are modal and therefore presented sequentially.  It's super user friendly, actually.  click recenter - type x value - enter - type y value - enter.  done.